### ISSUE TITLE
make temporary tables mandatory

### ISSUE THREAD

**Issue by rcarpa** [rucio/rucio#6238]
Many parts of rucio have now two versions of the same code: using temporary tables and the old version, without temporary tables. 

Get rid of the code paths which don't use temporary tables.

**Comment by dynamic-entropy** [comment]
Hello, it would be nice to harden the test too. 
To prevent things like https://github.com/rucio/rucio/issues/6234

The existence of an alternative allowed us to revert by a config change.

**Comment by dchristidis** [comment]
Yes, but the existence of an alternative makes accurate testing harder. :wink:

**Comment by rcarpa** [comment]
@dynamic-entropy , the example you give only plays in favor of removing the code. The bug exists in the temporary table workflow only because it was fixed by https://github.com/rucio/rucio/pull/5676 in the code path without temporary tables, but not fixed in the other one. Also note, that there was a comment about needing to also fix the temporary table part in this same pull request, which was ignored and not implemented. 

Also, we did have mechanisms to test both code paths in tests. For example: 
https://github.com/rucio/rucio/blob/6de80fcde72da57f86d952d34a6585fd8b52f7ae/lib/rucio/tests/test_did.py#L127
